diff --git a/core/tasks/kernel.mk b/core/tasks/kernel.mk
index 1b062ee..eeec481 100644
--- a/core/tasks/kernel.mk
+++ b/core/tasks/kernel.mk
@@ -175,10 +175,10 @@ ifeq ($(TARGET_ARCH),arm)
         ifeq ($(HOST_OS),darwin)
             ARM_CROSS_COMPILE:=CROSS_COMPILE="$(ccache) $(ANDROID_BUILD_TOP)/prebuilt/darwin-x86/toolchain/$(TARGET_KERNEL_CUSTOM_TOOLCHAIN)/bin/arm-eabi-"
         else
-            ARM_CROSS_COMPILE:=CROSS_COMPILE="$(ccache) $(ANDROID_BUILD_TOP)/prebuilt/linux-x86/toolchain/$(TARGET_KERNEL_CUSTOM_TOOLCHAIN)/bin/arm-eabi-"
+            ARM_CROSS_COMPILE:=CROSS_COMPILE="$(ccache) $(ANDROID_BUILD_TOP)/prebuilt/linux-x86/toolchain/arm-cortex_a8-linux-gnueabi-linaro_4.7.4-$(TARGET_LINARO_TOOLCHAIN_DATE)/bin/arm-cortex_a8-linux-gnueabi-"
         endif
     else
-        ARM_CROSS_COMPILE:=CROSS_COMPILE="$(ccache) $(ARM_EABI_TOOLCHAIN)/arm-eabi-"
+        ARM_CROSS_COMPILE:=CROSS_COMPILE="$(ccache) $(ANDROID_BUILD_TOP)/prebuilt/linux-x86/toolchain/arm-cortex_a8-linux-gnueabi-linaro_4.7.4-$(TARGET_LINARO_TOOLCHAIN_DATE)/bin/arm-cortex_a8-linux-gnueabi-"
     endif
     ccache = 
 endif
