From bd463fa7029b6d56cbb409356c79e3d62a189d56 Mon Sep 17 00:00:00 2001
From: David Eddlemon <DavidEddlemon@Hotmail.com>
Date: Mon, 19 Aug 2013 23:55:34 +0400
Subject: [PATCH] Strict Aliasing Fix

---
 Android.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Android.mk b/Android.mk
index 5eaa892..3dc2e3a 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,6 +1,9 @@
 LOCAL_PATH:= $(call my-dir)
 
+LOCAL_CFLAGS += -fno-strict-aliasing
+
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES:= ping.c ping_common.c
 LOCAL_MODULE := ping
+LOCAL_CFLAGS += -fno-strict-aliasing
 include $(BUILD_EXECUTABLE)
-- 
1.8.1.6

