From bb3bfe9965f28be23a20471fe0e2838e90aee350 Mon Sep 17 00:00:00 2001
From: David Eddlemon <DavidEddlemon@Hotmail.com>
Date: Tue, 20 Aug 2013 00:06:24 +0400
Subject: [PATCH] Strict Aliasing Fix

---
 Android.mk | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/Android.mk b/Android.mk
index f418740..ff596f4 100644
--- a/Android.mk
+++ b/Android.mk
@@ -3,6 +3,8 @@ LOCAL_PATH:= $(call my-dir)
 ###################### libssh ######################
 include $(CLEAR_VARS)
 
+LOCAL_CFLAGS += -fno-strict-aliasing
+
 LOCAL_MODULE_TAGS := optional
 
 LOCAL_SRC_FILES := \
@@ -39,7 +41,8 @@ LOCAL_SHARED_LIBRARIES += libssl libcrypto libdl libz
 
 LOCAL_MODULE := libssh
 
-LOCAL_CFLAGS+=-O3
+LOCAL_CFLAGS += -O3 -fno-strict-aliasing
+LOCAL_CFLAGS += -fno-strict-aliasing
 
 include $(BUILD_SHARED_LIBRARY)
 
@@ -49,6 +52,8 @@ include $(CLEAR_VARS)
 
 LOCAL_MODULE_TAGS := optional
 
+LOCAL_CFLAGS += -fno-strict-aliasing
+
 LOCAL_SRC_FILES := \
     ssh.c readconf.c clientloop.c sshtty.c \
     sshconnect.c sshconnect1.c sshconnect2.c mux.c \
@@ -169,3 +174,5 @@ LOCAL_MODULE := start-ssh
 LOCAL_MODULE_CLASS := EXECUTABLES
 LOCAL_SRC_FILES := start-ssh
 include $(BUILD_PREBUILT)
+
+LOCAL_CFLAGS += -fno-strict-aliasing
-- 
1.8.1.6

