diff --git a/src/java/com/android/internal/telephony/MccTable.java b/src/java/com/android/internal/telephony/MccTable.java
index c00b4a4..8d36ffe 100644
--- a/src/java/com/android/internal/telephony/MccTable.java
+++ b/src/java/com/android/internal/telephony/MccTable.java
@@ -204,9 +204,7 @@ public final class MccTable
                 setTimezoneFromMccIfNeeded(context, mcc);
                 locale = getLocaleFromMcc(context, mcc);
             }
-            if (fromServiceState) {
-                setWifiCountryCodeFromMcc(context, mcc);
-            } else {
+            if (!fromServiceState) {
                 // from SIM
                 try {
                     Configuration config = new Configuration();
