diff --git a/src/com/android/providers/media/MediaProvider.java b/src/com/android/providers/media/MediaProvider.java
index 538ccc2..91560f4 100755
--- a/src/com/android/providers/media/MediaProvider.java
+++ b/src/com/android/providers/media/MediaProvider.java
@@ -5244,15 +5244,14 @@ public class MediaProvider extends ContentProvider {
             } else if (EXTERNAL_VOLUME.equals(volume)) {
                 if (Environment.isExternalStorageRemovable()) {
                     String path = mExternalStoragePaths[0];
+                    // For Kitkat we know this will fail but check anyway.
                     int volumeID = FileUtils.getFatVolumeId(path);
-                    //if (LOCAL_LOGV) Log.v(TAG, path + " volume ID: " + volumeID);
-                    Log.e(TAG, path + " volume ID: " + volumeID);
 
-                    // In case of a non-FAT filesystem, try to get the UUID
                     if (volumeID == -1) {
-                        volumeID = FileUtils.getVolumeUUID(path);
-                        // if (LOCAL_LOGV) Log.v(TAG, path + " UUID: " + volumeID);
-                        Log.e(TAG, path + " UUID: " + volumeID);
+                        // HACK: build the real mount point (NOTE: /mnt/media_rw must be 770)
+                        String realPath = "/mnt/media_rw/" + path.substring(path.lastIndexOf('/')+1);
+                        volumeID = FileUtils.getFatVolumeId(realPath);
+                        Log.d(TAG, realPath + " volume ID: " + volumeID);
                     }
 
                     // Must check for failure!
