From b4a4b0bed52f30bae67b1c43b31a7c9001975f63 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sat, 24 Aug 2013 18:42:29 -0700
Subject: [PATCH] Remove -Werror flag

---
 Android.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index 816344a..4658bcb 100644
--- a/Android.mk
+++ b/Android.mk
@@ -28,8 +28,6 @@ ifneq ($(TARGET_USE_CUSTOM_SECOND_LUN_NUM),)
 common_cflags += -DCUSTOM_SECOND_LUN_NUM=$(TARGET_USE_CUSTOM_SECOND_LUN_NUM)
 endif
 
-common_cflags += -Werror
-
 common_src_files := \
 	VolumeManager.cpp \
 	CommandListener.cpp \
-- 
1.8.1.6

